ethereum/dbstack/db_stack_test.go

```diff
@@ -54,3 +55,44 @@ func TestDBStackPushPeekPop(t *testing.T) {
 	require.NoError(t, err)
 	assert.Len(t, miniHeaders, expectedLen)
 }
+
+func TestDBStackReset(t *testing.T) {
+	meshDB, err := meshdb.New("/tmp/leveldb_testing/" + uuid.New().String())
+	require.NoError(t, err)
+	stack, err := New(meshDB, 10)
```

Comment:
We should improve this test by doing more than one push/pop. We should also include a case where the same miniheader is pushed and popped more than once. We should also include a test where we call both `Checkpoint` and `Reset`.